include/wx/dc.h

```diff
@@ -1405,6 +1405,78 @@ class WXDLLIMPEXP_CORE wxDCTextColourChanger
     wxDECLARE_NO_COPY_CLASS(wxDCTextColourChanger);
 };
 
+// ----------------------------------------------------------------------------
+// helper class: you can use it to temporarily change the DC text background colour and
+// restore it automatically when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCTextBgColourChanger
+{
+public:
+    wxDCTextBgColourChanger(wxDC& dc) : m_dc(dc), m_colBgOld() { }
+
+    wxDCTextBgColourChanger(wxDC& dc, const wxColour& col) : m_dc(dc)
+    {
+        Set(col);
+    }
+
+    ~wxDCTextBgColourChanger()
+    {
+        if ( m_colBgOld.IsOk() )
+            m_dc.SetTextBackground(m_colBgOld);
+    }
+
+    void Set(const wxColour& col)
+    {
+        if ( !m_colBgOld.IsOk() )
+            m_colBgOld = m_dc.GetTextBackground();
+        m_dc.SetTextBackground(col);
+    }
+
+private:
+    wxDC& m_dc;
+
+    wxColour m_colBgOld;
+
+    wxDECLARE_NO_COPY_CLASS(wxDCTextBgColourChanger);
+};
+
+// ----------------------------------------------------------------------------
+// helper class: you can use it to temporarily change the DC text background mode and
+// restore it automatically when the object goes out of scope
+// ----------------------------------------------------------------------------
+
+class WXDLLIMPEXP_CORE wxDCTextBgModeChanger
+{
+public:
+    wxDCTextBgModeChanger(wxDC& dc) : m_dc(dc), m_modeOld(wxBRUSHSTYLE_INVALID) { }
+
+    wxDCTextBgModeChanger(wxDC& dc, int mode) : m_dc(dc)
+    {
+        Set(mode);
+    }
+
+    ~wxDCTextBgModeChanger()
+    {
+        if ( wxBRUSHSTYLE_INVALID != m_modeOld )
```

Comment:
I believe wxWidgets does not use coding style with the constant on the left in conditions of `if` statements. 

Applies also to `Set()`.